app/code/Magento/InventorySales/Plugin/CatalogInventory/StockManagement/ProcessBackItemQtyPlugin.php

```diff
@@ -49,12 +61,16 @@ public function __construct(
         GetSkusByProductIdsInterface $getSkusByProductIds,
         StockByWebsiteIdResolver $stockByWebsiteIdResolver,
         ReservationBuilderInterface $reservationBuilder,
-        AppendReservationsInterface $appendReservations
+        AppendReservationsInterface $appendReservations,
+        ProductRepositoryInterface $productRepository,
```

Comment:
ProductRepositoryInterface is a heavyweight object and we can't use it in lightweight plugins.

Taking into account that there is no other valid options to do that using Catalog APIs, 
we need to introduce new Service for catalog module - 
`GetProductTypeBySkuInterface`

Reply:
@maghamed  it seems like in this case we need to introduce one more service to get product type by product id, because in Magento\InventorySales\Plugin\CatalogInventory\StockManagement\ProcessBackItemQtyPlugin::aroundBackItemQty we have no sku in arguments.

Reply:
@maghamed done